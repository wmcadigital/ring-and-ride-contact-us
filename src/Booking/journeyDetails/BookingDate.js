import { useFormState } from "react-final-form";

import FormSection from "../../common/FormSection";
import Question from "../../common/Question";
import ProgressIndicator from "../../common/ProgressIndicator";
import DateInput from "../../common/DateInput";
import getSectionPositionInfo from "../getSectionPosition";
import getAboutJourneySectionName from "./getAboutJourneySectionName";

const BookingDate = () => {
  const stateApi = useFormState();

  const error = stateApi.submitFailed ? stateApi.errors?.bookingDate : null;

  return (
    <FormSection>
      <ProgressIndicator
        sectionPosition={getSectionPositionInfo("2", stateApi)}
        sectionName={getAboutJourneySectionName(stateApi)}
      />
      <Question text="What date is the booking for?" />
      <DateInput
        dayFieldName="bookingDateDay"
        monthFieldName="bookingDateMonth"
        yearFieldName="bookingDateYear"
        label="For example, 3 7 1985"
        error={error}
      />
    </FormSection>
  );
};

export default BookingDate;
