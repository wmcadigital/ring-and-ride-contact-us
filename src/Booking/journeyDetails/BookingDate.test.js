import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import BookingDate from "./BookingDate";

jest.mock("react-final-form");

describe("BookingDate", () => {
  it("individual is filling out form for themselves", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "mySelf",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<BookingDate />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("individual is filling out form for someone else", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "behalfSomeone",
        firstName: "John",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<BookingDate />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("individual is filling out form for a group", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "behalfGroup",
      },
    }));
    const renderer = createRenderer();

    renderer.render(<BookingDate />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders with an error", () => {
    useFormState.mockImplementationOnce(() => ({
      values: {
        bookingParty: "mySelf",
        firstName: "John",
      },
      submitFailed: true,
      errors: { bookingDate: "Required" },
    }));
    const renderer = createRenderer();

    renderer.render(<BookingDate />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
