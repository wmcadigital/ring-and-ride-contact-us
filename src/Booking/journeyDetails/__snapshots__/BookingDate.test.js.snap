// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingDate individual is filling out form for a group 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan their journey"
    sectionPosition="Section 3 of 5"
  />
  <Question
    text="What date is the booking for?"
  />
  <DateInput
    dayFieldName="bookingDateDay"
    error={null}
    label="For example, 3 7 1985"
    monthFieldName="bookingDateMonth"
    yearFieldName="bookingDateYear"
  />
</FormSection>
`;

exports[`BookingDate individual is filling out form for someone else 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan their journey"
    sectionPosition="Section 2 of 4"
  />
  <Question
    text="What date is the booking for?"
  />
  <DateInput
    dayFieldName="bookingDateDay"
    error={null}
    label="For example, 3 7 1985"
    monthFieldName="bookingDateMonth"
    yearFieldName="bookingDateYear"
  />
</FormSection>
`;

exports[`BookingDate individual is filling out form for themselves 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan your journey"
    sectionPosition="Section 2 of 4"
  />
  <Question
    text="What date is the booking for?"
  />
  <DateInput
    dayFieldName="bookingDateDay"
    error={null}
    label="For example, 3 7 1985"
    monthFieldName="bookingDateMonth"
    yearFieldName="bookingDateYear"
  />
</FormSection>
`;

exports[`BookingDate renders with an error 1`] = `
<FormSection>
  <ProgressIndicator
    sectionName="Plan your journey"
    sectionPosition="Section 2 of 4"
  />
  <Question
    text="What date is the booking for?"
  />
  <DateInput
    dayFieldName="bookingDateDay"
    error="Required"
    label="For example, 3 7 1985"
    monthFieldName="bookingDateMonth"
    yearFieldName="bookingDateYear"
  />
</FormSection>
`;
